src/main/java/com/cloudbees/dockerpublish/DockerBuilder.java

```diff
@@ -77,32 +80,124 @@ public DockerBuilder(String repoName, String repoTag, boolean skipPush, boolean
         this.skipTagLatest = skipTagLatest;
     }
 
-    public String getRepoName() {return repoName; }
-    public String getRepoTag() {  return repoTag; }
-    public boolean isSkipPush() { return skipPush;}
-    public boolean isSkipBuild() { return skipBuild;}
-    public boolean isSkipDecorate() { return skipDecorate;}
-    public boolean isSkipTagLatest() { return skipTagLatest;}
-    public boolean isNoCache() { return noCache;}
-    public boolean isForcePull() { return forcePull;}
-    public String getDockerfilePath() { return dockerfilePath; }
+    @DataBoundConstructor
+    public DockerBuilder(DockerServerEndpoint server, DockerRegistryEndpoint registry, String repoName) {
```

Comment:
Since `server` and `registry` have obvious default values (new instances of those classes with null/blank `uri`/`url`/`credentialsId`), I would recommend using `@DataBoundSetter` and initializing the fields with those defaults. Generally a `@DataBoundConstructor` should only list parameters which are obviously mandatory for the user to specify. Simplifies the constructor, which makes it easier to keep compatibility as the class evolves, and is useful for scripting calls (Workflow, YAML project, etc.).